### ISSUE TITLE
passing a `nil` metadata into Publish crashes

### ISSUE THREAD

**Issue by dmacthedestroyer** [infobloxopen/atlas-pubsub#35]
```
publisher := grpc.NewPublisher(...)
publisher.Publish(ctx, msg, nil)
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x13cfe11]

goroutine 90 [running]:
github.com/infobloxopen/atlas-pubsub/aws.decodeMessageAttributes(0xc4200bf708, 0xc420330c40)
	/Users/dmcdonald/go/src/github.com/infobloxopen/atlas-pubsub/aws/aws.go:341 +0x191
github.com/infobloxopen/atlas-pubsub/aws.(*awsSubscriber).pull(0xc4203e68c0, 0x16a4f00, 0xc4202ad240, 0xc420428960, 0xc4204289c0)
	/Users/dmcdonald/go/src/github.com/infobloxopen/atlas-pubsub/aws/subscriber.go:291 +0x336
github.com/infobloxopen/atlas-pubsub/aws.(*awsSubscriber).Start.func1(0xc420428960, 0xc4203e68c0, 0xc420356f40, 0xc4204289c0, 0x16a4f00, 0xc4202ad240)
	/Users/dmcdonald/go/src/github.com/infobloxopen/atlas-pubsub/aws/subscriber.go:82 +0x108
created by github.com/infobloxopen/atlas-pubsub/aws.(*awsSubscriber).Start
	/Users/dmcdonald/go/src/github.com/infobloxopen/atlas-pubsub/aws/subscriber.go:65 +0x127
exit status 2
```

**Comment by dmacthedestroyer** [comment]
this was caused by changes from pr #28

**Comment by dmacthedestroyer** [comment]
addressed in PR #36
